### DISCUSSION TITLE
Error message processing/collecting errors

### DISCUSSION THREAD

**Discussion by Busschers** [microsoft/alguidelines#240] [BC Patterns]
When I searched for the best way to handle collectable errors I found two articles, one in the Patterns Archive of Alguidelines and one on Microsoft Learn.

Alguidelines version:
https://alguidelines.dev/docs/navpatterns/patterns/error-message-processing/

Microsoft Learn version:
https://learn.microsoft.com/en-us/dynamics365/business-central/dev-itpro/developer/devenv-error-collection

I think the Microsoft Learn version describes the modern way of collecting errors, am I right?

If that is the case I was thinking about desribing this way in the 'Design Patterns' section of AL guidelines. What is a proper way of doing this, just a link to the microsoft learn site or more a copy/paste way of all the text and images?

**Comment by waldo1001** [comment]
This for sure is one that needs some guidelines ;-).

May be this is one that should be added to the resources-list:
https://www.youtube.com/watch?v=D8233xMjVog&list=PLI1l3dMI8xlDM9onioMWUyMSCiFs_mMWw&index=27

**Comment by JeremyVyska** [comment]
I feel like we can add a lot more to ErrorHandling guidelines, but I'd rather have *something* to get us started and build up over time.  So this feels like a great start.
